### PR TITLE
test(raft): improve priority time test for readability

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

Follow up of PR review #7336
Improve the test for readability : use mocks to verify that the timers are triggered in the expected order.

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: atomix/cluster/src/test/java/io/atomix/raft/impl/PriorityElectionTimerTest.java [hunk 1]
```diff
@@ -54,7 +57,9 @@ public void shouldLowerPriorityNodeEventuallyStartsAnElection() {
   @Test
   public void shouldHighPriorityNodeStartElectionFirst() {
     // given
-    final AtomicBoolean highPrioElectionTriggered = new AtomicBoolean();
+    final AtomicBoolean highPrioElectionTriggered = spy(new AtomicBoolean());
```
Comment (Zelldon) [inline atomix/cluster/src/test/java/io/atomix/raft/impl/PriorityElectionTimerTest.java:60]: Interesting. I thought you just use `mock(Runnable.class)`. I think it is not necessary to use the AtomicBoolean, but if you prefer it like that, I'm fine.

**Review by ChrisKujawa — APPROVED** [review]
Thanks :)

File: atomix/cluster/src/test/java/io/atomix/raft/impl/PriorityElectionTimerTest.java (hunk 1 above)
Comment (deepthidevaki) [reply atomix/cluster/src/test/java/io/atomix/raft/impl/PriorityElectionTimerTest.java:60]: I wanted to wait until it is triggered atleast once. When using AtomicBoolean, I can use it's get method to wait. I found it better than adding timeouts to verify.

Comment (Zelldon) [reply atomix/cluster/src/test/java/io/atomix/raft/impl/PriorityElectionTimerTest.java:60]: You could also things like `inOrder.verify(mock, timeout(100)).run();`

**Pull request merged**